static/app/views/ddm/optInModal.tsx

```diff
@@ -0,0 +1,172 @@
+import {useCallback, useRef} from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import HeroImg from 'sentry-images/spot/custom-metrics-opt-in-modal-hero.png';
+
+import {type ModalRenderProps, openModal} from 'sentry/actionCreators/modal';
+import {Button, type ButtonProps, LinkButton} from 'sentry/components/button';
+import {IconClose} from 'sentry/icons/iconClose';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Organization} from 'sentry/types';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+import useOrganization from 'sentry/utils/useOrganization';
+import {OrganizationContext} from 'sentry/views/organizationContext';
+
+export function openMetricsOptInModal(organization: Organization) {
+  return openModal(
+    deps => (
+      <OrganizationContext.Provider value={organization}>
+        <OptInModal
+          {...deps}
+          closeModal={() => {
+            localStorage.setItem('sentry:metrics-opt-in-modal-dismissed', 'true');
+            deps.closeModal();
+          }}
+        />
+      </OrganizationContext.Provider>
+    ),
+    {modalCss}
```

Comment:
nit: I would disable closing on clicking the backdrop for this modal as it can easily happen by mistake.
```suggestion
    {modalCss, closeEvents: 'none'}
```